docs/future/migration.md

```diff
@@ -0,0 +1,372 @@
+# Migration Guide for Enzyme v2.x to v3.x
+
+The change from Enzyme v2.x to v3.x is a more significant change than in previous major releases,
+due to the fact that the internal implementation has been almost completely rewritten.
+
+The goal of this rewrite was to address a lot of the major issues that have plagued Enzyme since
+its initial release. It was also to simultaneously remove a lot of the dependence that Enzyme has
+on react internals, and to make enzyme more "pluggable", paving the way for Enzyme to be used
+with "React-like" libraries such as Preact and Inferno.
+
+We have done our best to make Enzyme v3 as API compatible with v2.x as possible, however there are
+a hand full of breaking changes that we decided we needed to make, intentionally, in order to
+support this new architecture.
+
+Airbnb has one of the largest Enzyme test suites, coming in at around 30,000 enzyme unit tests.
+After upgrading Enzyme to v3.x in Airbnb's code base, 99.6% of these tests succeeded with no
+modifications at all. Most of the tests that broke we found to be easy to fix, and some we found to
+actually be depending on what could arguably be considered a bug in v2.x, and the breakage was
+desired.
+
+In this guide, we will go over a couple of the most common breakages that we ran into, and how to
+fix them. Hopefully this will make your upgrade path that much easier.
+
+
+## Configuring your Adapter
+
+Enzyme now has an "Adapter" system. This means that you now need to install Enzyme along with
+another module that provides the Adapter that tells Enzyme how to work with your version of React
+(or whatever other react-like library you are using).
+
+At the time of writing this, Enzyme publishes "officially supported" adapters for React 0.13.x,
+0.14.x, 15.x, and 16.x. These adapters are npm packages of the form `enzyme-adapter-react-{{version}}`.
+
+You will want to configure Enzyme with the adapter you'd like to use before using enzyme in your
+tests. The way to do this is whith `Enzyme.configure(...)`. For example, if your project depends
+on React 16, you would want to configure Enzyme this way:
+
+```js
+import Enzyme from 'enzyme';
```

Comment:
aesthetically i kinda prefer this way for some reason, but i don't think i can justify it in any way :P